### PR TITLE
Modify CustomerStore to support searchCustomers action

### PR DESCRIPTION
**Pull request by iamgabrielma**
Part of https://github.com/woocommerce/woocommerce-ios/issues/7741 , branched from https://github.com/woocommerce/woocommerce-ios/pull/7780 so that should be merged first.

### Description

This PR modifies the existing `CustomerStore` and `CustomerAction` to implement both `CustomerRemote`, and `WCAnalyticsCustomerRemote` as well as their respective actions under the same class.

It also provides a temporary implementation for the `searchCustomers()` action, which returns a `Result<Void, Error>` type to a completion handler, this performs the customer search by name, and is intended to be saved on Core Data in future iterations as "search results".

### Changes:
- Declare a `customerRemote` and a `searchRemote` under `CustomerStore`.
- Implements the `searchCustomers()` action, which performs the customer search by keyword. Only searches by name.
- Modifies the "magnifying glass" button in "Orders" > "+" > "Create new Order" > "+ Add Customer" that calls the action.

### Testing instructions:

1. The current implementation assumes that you have at least one customer in your store, and that customer name contains the word "hello", if that's not the case, you can change this line to the keyword you prefer:

https://github.com/woocommerce/woocommerce-ios/blob/0e33748fcdb9cf46506b6ddef2e6f923a5e6e04a/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/CustomerSection/CreateOrderAddressFormViewModel.swift#L108

3. Go to "Orders" > "+" > "Create new Order" > "+ "Add Customer" and tap in the "Magnifying glass" icon at the top of the screen

4. Confirm in Proxyman or any other tool that we send a GET HTTP request to /wc-analytics/customers?search={your-keyword} and that returns a 200 response, example:

```
https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/121710041/rest-api/?json=true&path=/wc-analytics/customers%26_method%3Dget&query=%7B%22search%22%3A%22hello%22%7D
```

```
{
  "data": [
    {
      "id": 99,
      "user_id": 4,
      "username": "hello.merch",
      "name": "hello merch",
      "email": "gabriel.maldonado+testuser20221004@automattic.com",
      "country": "US",
      "city": "soller",
      "state": "CA",
      "postcode": "01101",
      "date_registered": "2022-10-04T08:36:49",
      "date_last_active": "2022-10-04T08:36:49",
      "date_last_order": "2022-10-04 08:36:49",
      "orders_count": 1,
      "total_spend": 1,
      "avg_order_value": 1,
      "date_registered_gmt": "2022-10-04T08:36:49",
      "date_last_active_gmt": "2022-10-04T08:36:49",
      "_links": {
        "customer": [
          {
            "href": "https://mywootestingstore.mystagingwebsite.com/wp-json/wc-analytics/customers/99"
          }
        ],
        "collection": [
          {
            "href": "https://mywootestingstore.mystagingwebsite.com/wp-json/wc-analytics/customers"
          }
        ]
      }
    }
  ]
}
```

### CODE REVIEW

File: Yosemite/Yosemite/Actions/WCAnalyticsCustomerAction.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public enum WCAnalyticsCustomerAction: Action {
+
+    /// Retrieves a collection of WCAnalyticsCustomer from a site
+    ///
+    ///- `siteID`: The site for which customers should be fetched.
+    ///- `keyword`: Keyword to perform the search for WCAnalyticsCustomer to be fetched.
+    ///- `onCompletion`: Invoked when the operation finishes.
+    ///     - `result.success(Customer)`: The Customer object
+    ///     - `result.failure(Error)`: Error fetching Customer
+    case retrieveCustomers(
```
Comment (joshheald) [inline Yosemite/Yosemite/Actions/WCAnalyticsCustomerAction.swift]: `searchCustomers` would be more descriptive of the action

File: Networking/Networking/Remote/WCAnalyticsCustomerRemote.swift [hunk 2]
```diff
@@ -9,18 +9,19 @@ public class WCAnalyticsCustomerRemote: Remote {
     ///     - name: Name of the customer that will be retrieved
     ///     - completion: Closure to be executed upon completion.
     ///
-    func retrieveCustomersByName(for siteID: Int64, with name: String, completion: @escaping (Result<[WCAnalyticsCustomer], Error>) -> Void) {
+    public func retrieveCustomersByName(for siteID: Int64, with name: String, completion: @escaping (Result<[WCAnalyticsCustomer], Error>) -> Void) {
```
Comment (joshheald) [inline Networking/Networking/Remote/WCAnalyticsCustomerRemote.swift]: `searchCustomersByName`

File: Yosemite/Yosemite/Model/Model.swift [hunk 3]
```diff
@@ -147,6 +147,7 @@ public typealias PrintingResult = Hardware.PrintingResult
 public typealias CardPresentReceiptParameters = Hardware.CardPresentReceiptParameters
 public typealias CardPresentTransactionDetails = Hardware.CardPresentTransactionDetails
 public typealias StripeAccount = Networking.StripeAccount
+public typealias WCAnalyticsCustomer = Networking.WCAnalyticsCustomer
```
Comment (iamgabrielma) [inline Yosemite/Yosemite/Model/Model.swift:150]: TODO: remove.

Comment (iamgabrielma) [inline Yosemite/Yosemite/Model/Model.swift]:
```suggestion

```

**Review by joshheald — APPROVED** [review]
Gone through in detail during pairing.

Looks good, once the test fix from the base PR is merged in. :shipit:

**Pull request merged**